### PR TITLE
Update CoreFX dependencies to rc3-24102-00

### PR DESCRIPTION
**Pull request by tmat**
Consuming the latest CoreFX packages uncovered a bug in nuget, which has apparently been fixed in 3.5.0.1246. See https://github.com/NuGet/Home/issues/2710

### CODE REVIEW

**Comment by tmat** [comment]
@jasonmalinowski @davkean 
@dotnet/roslyn-infrastructure

**Comment by tmat** [comment]
@dotnet-bot retest prtest/lin/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_lin_dbg_unit32/6804/
// Retest reason:

**Comment by tmat** [comment]
@dotnet-bot retest mac please

**Comment by tmat** [comment]
@dotnet-bot retest vsi please

**Comment by tmat** [comment]
@dotnet-bot retest prtest/mac/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_mac_dbg_unit32/5334/
// Retest reason:

**Comment by tmat** [comment]
@dotnet-bot retest prtest/lin/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_lin_dbg_unit32/6863/
// Retest reason:

**Comment by tmat** [comment]
@dotnet/roslyn-compiler @dotnet/roslyn-infrastructure Looks good?

**Comment by tmat** [comment]
test vsi please

File: src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/project.json [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+    "Microsoft.NETCore.Portable.Compatibility": "1.0.1-rc3-24102-00",
+    "NETStandard.Library": "1.5.0-rc3-24102-00",
+    "System.Xml.XmlDocument": "4.0.1-rc3-24102-00",
+    "System.Xml.XmlSerializer": "4.0.11-rc3-24102-00"
   },
   "frameworks": {
     "dnxcore50": { }
+  },
+  "runtimes": {
+    "win7-x64": {},
```
Comment (jasonmalinowski) [inline src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/project.json]: What about x86?

**Comment by jasonmalinowski** [comment]
:+1:, at least for the parts I understand (project.json updates, build updates.) Can't speak to the code changes. Do we not build the compiler generators for x86? What happens for somebody on a 32-bit OS?

**Comment by tmat** [comment]
I don't know what happens. Most of our project.json files only have -x64. Perhaps that's wrong?

**Comment by tmat** [comment]
test mac please

**Comment by tmat** [comment]
retest vsi please

**Comment by tmat** [comment]
retest vsi please

**Comment by tannergooding** [comment]
@tmat, @jasonmalinowski. I'm pretty sure the runtimes are only `x64` because there isn't a CoreCLR that targets `x86` right now (only desktop). I think this is more of nuget limitation than any lack of support for said platform on our end (but maybe @agocke can confirm).

**Comment by agocke** [comment]
@tannergooding @tmat There is an x86 runtime now (there wasn't before). There shouldn't be any limitation to running on the x86 version.

As to what happens if you build on an x86 machine right now: it won't work. In other news, we also take a dependency on this newfangled invention called "the internet." :trollface:

**Comment by tmat** [comment]
retest vsi please

**Comment by tmat** [comment]
retest lib please

**Comment by tmat** [comment]
retest lin please

**Comment by tmat** [comment]
retest linux please

**Comment by jasonmalinowski** [comment]
D'oh, wrong PR. Sorry for the kick. :-(

**Comment by tmat** [comment]
retest roslyn_prtest_win_vsi3 please

**Pull request merged**